compileopts/target.go

```diff
@@ -181,15 +181,28 @@ func LoadTarget(options *Options) (*TargetSpec, error) {
 		case "arm64":
 			llvmarch = "aarch64"
 		case "arm":
-			switch options.GOARM {
-			case "5":
-				llvmarch = "armv5"
-			case "6":
-				llvmarch = "armv6"
-			case "7":
-				llvmarch = "armv7"
-			default:
-				return nil, fmt.Errorf("invalid GOARM=%s, must be 5, 6, or 7", options.GOARM)
+			armOpts := strings.Split(options.GOARM, ",")
+			if len(armOpts) == 1 || len(armOpts) == 2 {
+				switch armOpts[0] {
+				case "5":
+					llvmarch = "armv5"
+				case "6":
+					llvmarch = "armv6"
+				case "7":
+					llvmarch = "armv7"
+				default:
+					return nil, fmt.Errorf("invalid GOARM=%s, must be 5, 6, or 7", options.GOARM)
+				}
+				if len(armOpts) == 2 {
+					switch armOpts[1] {
+					case "softfloat":
+						llvmarch += "sf"
+					case "hardfloat":
+						llvmarch += "hf"
+					default:
+						return nil, fmt.Errorf("invalid extension GOARM=%s, must be softfloat or hardfloat", options.GOARM)
+					}
```

Comment:
I don't think this is correct, I think you need to end the triple with `-gnueabi` or `-gnueabihf` instead (see the definition of `target` a few lines below - GitHub doesn't let me comment there).

Reply:
Isn't that part being substituted further down the line anyways? Afaik the armvX part gets turned back into just arm as LLVM doesn't explicitly check the arm32 version based on the first element of the target triple.

Reply:
I've found that whatever you put after `armv7` or `armv6` appears to be ignored entirely.
What I mean is that this piece of code needs updating to have the correct suffix:

```go
		// Target triples (which actually have four components, but are called
		// triples for historical reasons) have the form:
		//   arch-vendor-os-environment
		target := llvmarch + "-" + llvmvendor + "-" + llvmos
		if options.GOOS == "windows" {
			target += "-gnu"
		} else if options.GOARCH == "arm" {
			target += "-gnueabihf"
		}
```

Here you'll need to modify the environment part (`-gnueabihf`) to something else to disable floating point instructions (which will automatically result in a softfloat fallback).

You may find this page useful to figure out the correct flags: https://embeddedartistry.com/blog/2017/10/11/demystifying-arm-floating-point-compiler-options/

Reply:
I updated the PR to change that part of the target triple now but also change the flags further down. I'm actually not too sure still what the correct subset would be that still supports the majority of chips and still gives decent performance. It looks like most of the FPU stuff is optional anyways so what would be a sane subset when someone explicitly requests hardfloat support for a given arm version?